src/com/amplitude/api/DatabaseHelper.java

```diff
@@ -46,20 +51,43 @@
 
     private File file;
 
-    private static AmplitudeLog logger = AmplitudeLog.getLogger();
+    private static final AmplitudeLog logger = AmplitudeLog.getLogger();
+
+    @Deprecated
+    static DatabaseHelper getDatabaseHelper(Context context) {
+        return getDatabaseHelper(context, null);
+    }
 
-    static synchronized DatabaseHelper getDatabaseHelper(Context context) {
-        if (instance == null) {
-            instance = new DatabaseHelper(context.getApplicationContext());
+    static synchronized DatabaseHelper getDatabaseHelper(Context context, String instance) {
+        if (TextUtils.isEmpty(instance)) {
+            logger.w(TAG, "Should not be calling getDatabaseHelper() with an empty instance name");
+            instance = Constants.DEFAULT_INSTANCE;
         }
-        return instance;
+
+        DatabaseHelper dbHelper = instances.get(instance);
+        if (dbHelper == null) {
+            if (instance.equals(Constants.DEFAULT_INSTANCE)) {
+                dbHelper = new DatabaseHelper(context.getApplicationContext());
+            } else {
+                dbHelper = new DatabaseHelper(context.getApplicationContext(), instance);
+            }
+            instances.put(instance, dbHelper);
+        }
+        return dbHelper;
     }
 
+    // Maintain backwards compatability, allow access to old original database file
```

Comment:
Since the method is private, this is not actually for backwards compatibility right? It's for testing? (And the comment probably makes more sense on the static method above.)